corehq/apps/user_importer/importer.py

```diff
@@ -484,13 +485,23 @@ def create_or_update_commcare_users_and_groups(upload_domain, user_specs, upload
                         kwargs['is_account_confirmed'] = is_account_confirmed
                     user = CommCareUser.create(domain, username, password, created_by=upload_user,
                                                created_via=USER_CHANGE_VIA_BULK_IMPORTER, commit=False, **kwargs)
-                    log_user_create = True
                     status_row['flag'] = 'created'
 
+                user_change_logger = UserChangeLogger(upload_domain, user=user, is_new_user=not bool(user_id),
+                                                      changed_by_user=upload_user,
+                                                      changed_via=USER_CHANGE_VIA_BULK_IMPORTER)
+                if status_row['row'].get('password') == 'REDACTED':
+                    user_change_logger.add_change_message(_("Password Reset"))
+
                 if phone_number:
-                    user.add_phone_number(_fmt_phone(phone_number), default=True)
+                    fmt_phone_number = _fmt_phone(phone_number)
+                    # always call this to set phone number as default if needed
+                    user.add_phone_number(fmt_phone_number, default=True)
+                    if fmt_phone_number not in user.phone_numbers:
+                        user_change_logger.add_change_message(_(f"Added phone number {fmt_phone_number}"))
```

Comment:
Corrected later to check if number is already present before updating
https://github.com/dimagi/commcare-hq/pull/29935/commits/632e442311538caa417cbbd01f19d8486d2231bb